pkg/infinispan/cluster.go

```diff
@@ -241,6 +251,99 @@ func (c Cluster) CreateCacheWithTemplateName(cacheName, templateName, podName st
 	return validateResponse(rsp, reason, err, "creating cache with template", http.StatusOK)
 }
 
+// CreateCache create cache on the pod `podName`
+func (c Cluster) CreateCache(cacheName, jsonConfig, podName string) error {
+	path := fmt.Sprintf("%s/caches/%s", consts.ServerHTTPBasePath, cacheName)
+	rsp, err, reason := c.Client.Post(podName, path, jsonConfig, Headers{"Content-Type": "application/json"})
+	return validateResponse(rsp, reason, err, "creating cache", http.StatusOK)
+}
+
+// GetCacheConfig returns the cache configuration
+func (c Cluster) GetCacheConfig(cacheName, podName string) (cacheConfig string, err error) {
+	path := fmt.Sprintf("%s/caches/%s?action=config", consts.ServerHTTPBasePath, cacheName)
+	rsp, err, reason := c.Client.Get(podName, path, nil)
+	if err = validateResponse(rsp, reason, err, "getting cache config", http.StatusOK); err != nil {
+		return
+	}
+	defer func() {
+		cerr := rsp.Body.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+	var body json.RawMessage
+	if err = json.NewDecoder(rsp.Body).Decode(&body); err != nil {
+		return "", fmt.Errorf("unable to decode: %w", err)
+	}
+	return string(body), nil
+}
+
+// AddRemoteStore connects the target cluster to the source cluster using a Remote Store
+func (c Cluster) AddRemoteStore(cacheName, jsonConfig, podName string) error {
```

Comment:
`AddRemoteStore`, `IsSourceConnected`, `DisconnectSource` and `SyncData` are all very specific Rolling upgrades. It would be nice if we had another interface/implementation specific for Rolling upgrades to separate concerns better (something we haven't been great with so far).